pulsar-io/flume/pom.xml

```diff
@@ -106,7 +125,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>18.0</version>
+            <version>${guava.version}</version>
```

Comment:
Does the current Flume version support latest Guava version? There are multiple breaking changes between 18.0 and 31.0 .

Reply:
The tests passed, that's as much as I can tell.

Reply:
I believe that there must have been a reason why the version was pinned to 18.0 . I doubt that the tests in pulsar-io/flume validate the full functionality. @tuteng could you shade some light into this since you are the original author?

Reply:
@lhotari the best testing I could do is:
1. get flume's code. 
2. checkout d4fcab4f501d41597bc616921329a4339f73585e (last release), build. Build fails, with jdk 11 and jdk 8.
3. checkout `trunk` (some dependencies upgraded after the release but no dramatic changes). build succeeds. 
4. the guava dependency is brought in by `org.apache.flume:flume-ng-node` - so `cd flume/flume-ng-node` and run tests there. tests pass (with jdk8, some failed with jdk11)
5. force `<version>31.0.1-jre</version>` for guava in `flume-ng-node`. Build succeeds. `mvn dependency:tree` confirms that updated guava is used. The tests pass.

Reply:
great work in confirming the compatibility @dlg99